runtime-sdk/modules/evm/src/precompile/confidential.rs

```diff
@@ -58,53 +104,30 @@ pub(super) fn call_x25519_derive(
 #[allow(clippy::type_complexity)]
 fn decode_deoxysii_call_args(
     input: &[u8],
-) -> Result<([u8; KEY_SIZE], [u8; NONCE_SIZE], &[u8], &[u8]), PrecompileFailure> {
-    // Number of fixed words in the input (key, nonce word, two lengths; see
-    // comments in the precompiles).
-    const SLOTS: usize = 4;
-
-    if input.len() < SLOTS * WORD {
-        return Err(PrecompileFailure::Error {
-            exit_status: ExitError::Other("input length must be at least 128 bytes".into()),
-        });
-    }
-
-    let mut words = input.array_chunks::<WORD>().take(SLOTS);
-    let key = words.next().unwrap();
-    let nonce_word = words.next().unwrap();
-    let text_len = words.next().unwrap();
-    let ad_len = words.next().unwrap();
+) -> Result<([u8; KEY_SIZE], [u8; NONCE_SIZE], Vec<u8>, Vec<u8>), PrecompileFailure> {
+    let mut call_args = ethabi::decode(
```

Comment:
Does this break compatibility? Can we pretend nobody is using this yet?

Reply:
> Does this break compatibility?

It does, but we can definitely pretend that nobody is using this yet. I'm very confident that if we checked, we'd find that nobody is using this yet. Either way, it's testnet, and this new ABI is much easier to use.

Reply:
Agreed about the ease of use, not sure why we didn't use this immediately? This will also require fixing the documentation and library, in teh sapphire repo, will you do the fixes there as well?

Mainnet deployment is imminent so let's get this to Testnet quickly and make sure that Mainnet doesn't have the old version.

Reply:
We should also make sure this ABI is used in #1176.

Reply:
> This will also require fixing the documentation and library, in teh sapphire repo, will you do the fixes there as well?

100% going to do that after this PR goes in and before the next upgrade.

> Agreed about the ease of use, not sure why we didn't use this immediately? 

Didn't think of it then, as I hadn't experienced first-hand how awful it was to write and maintain the Solidity assembly.

> We should also make sure this ABI is used in #1176.

I'll apply the changes to this one, too.

Reply:
> > We should also make sure this ABI is used in #1176.
> 
> I'll apply the changes to this one, too.

No need, I'll be doing this for the next rebase, and I've got cleanup in progress anyway.